app/src/main/java/org/ole/planet/myplanet/service/SyncManager.java

```diff
@@ -77,14 +77,18 @@ public void destroy() {
         NotificationUtil.cancel(context, 111);
         isSyncing = false;
         ourInstance = null;
+        try{
+            mRealm.close();
+            td.stop();
+        }catch (Exception e){}
         settings.edit().putLong("LastSync", new Date().getTime()).commit();
         if (listener != null) {
             listener.onSyncComplete();
         }
     }
-
+    Thread td;
```

Comment:
I think it would be better to execute the thread from activity or fragment running this function and close the thread on `onDestroy()`. Initializing the thread here can it cause the memory leak problem if
The activity or fragment holding the reference if this class is destroyed or configuration is changed because on both case android will create the new activity and send the old one in garbage collection However, the thread holds an implicit reference to the old Activity and prevents it from ever being reclaimed.

Reply:
No, worker thread does not interact with UI so it is not necessary to define them in activity or fragment, and we can close them when they are not required.
AutoSyncService is also using same syncmanager so we must define the thread in this sync manager file, not in activity 
And other thing if we move this to an activity and close the thread in onDestory() we'll end up with same app freeze issue, as the thread will be running on background when user tries to login.